pkg/streamingpromql/operators/range_vector_selector.go

```diff
@@ -111,17 +114,23 @@ func (m *RangeVectorSelector) fillBuffer(floats *types.FPointRingBuffer, histogr
 				return nil
 			}
 		case chunkenc.ValHistogram, chunkenc.ValFloatHistogram:
-			t, h := m.chunkIterator.AtFloatHistogram(nil)
-			if value.IsStaleNaN(h.Sum) || t < rangeStart {
+			t := m.chunkIterator.AtT()
+			if t < rangeStart {
 				continue
 			}
-
-			// We might append a sample beyond the range end, but this is OK:
-			// - callers of NextStepSamples are expected to pass the same RingBuffer to subsequent calls, so the point is not lost
-			// - callers of NextStepSamples are expected to handle the case where the buffer contains points beyond the end of the range
-			if err := histograms.Append(promql.HPoint{T: t, H: h}); err != nil {
-				return err
+			hPoint, _ := histograms.NextPoint()
+			if hPoint.H == nil {
+				hPoint.H = &histogram.FloatHistogram{}
+			}
```

Comment:
Is this necessary? Won't `m.chunkIterator.AtFloatHistogram` allocate a `FloatHistogram` if it needs it? 

I think this and the line below could be simplified to this:

```go
hPoint.T, hPoint.H = m.chunkIterator.AtFloatHistogram(hPoint.H)
```

Then we can also drop the `hPoint.T = t` a few lines later.

Reply:
Thanks, good catch. This was from previous testing/benchmarking trials.